crates/control/src/fall_state_estimation.rs

```diff
@@ -1,42 +1,66 @@
+use coordinate_systems::Robot;
+use linear_algebra::{vector, Vector2, Vector3};
 use serde::{Deserialize, Serialize};
 
 use context_attribute::context;
 use filtering::low_pass_filter::LowPassFilter;
 use framework::{AdditionalOutput, MainOutput};
 use types::{
-    fall_state::FallState,
-    motion_command::{Facing, FallDirection},
-    parameters::FallStateEstimationParameters,
-    sensor_data::{InertialMeasurementUnitData, SensorData},
+    cycle_time::CycleTime,
+    fall_state::{Direction, FallState, Kind, Side},
+    joints::Joints,
+    sensor_data::SensorData,
 };
 
 #[derive(Deserialize, Serialize)]
 pub struct FallStateEstimation {
-    roll_pitch_filter: LowPassFilter<Vector2<f32>>,
-    angular_velocity_filter: LowPassFilter<Vector3<f32>>,
-    linear_acceleration_filter: LowPassFilter<Vector3<f32>>,
+    roll_pitch_filter: LowPassFilter<Vector2<Robot>>,
+    angular_velocity_filter: LowPassFilter<Vector3<Robot>>,
+    linear_acceleration_filter: LowPassFilter<Vector3<Robot>>,
+    last_fall_state: FallState,
 }
 
 #[context]
 pub struct CreationContext {
-    fall_state_estimation: Parameter<FallStateEstimationParameters, "fall_state_estimation">,
+    linear_acceleration_low_pass_factor:
+        Parameter<f32, "fall_state_estimation.linear_acceleration_low_pass_factor">,
+    angular_velocity_low_pass_factor:
+        Parameter<f32, "fall_state_estimation.angular_velocity_low_pass_factor">,
+    roll_pitch_low_pass_factor: Parameter<f32, "fall_state_estimation.roll_pitch_low_pass_factor">,
 }
 
 #[context]
 pub struct CycleContext {
-    backward_gravitational_difference: AdditionalOutput<f32, "backward_gravitational_difference">,
-    filtered_angular_velocity: AdditionalOutput<Vector3<f32>, "filtered_angular_velocity">,
-    filtered_linear_acceleration: AdditionalOutput<Vector3<f32>, "filtered_linear_acceleration">,
-    filtered_roll_pitch: AdditionalOutput<Vector2<f32>, "filtered_roll_pitch">,
-    forward_gravitational_difference: AdditionalOutput<f32, "forward_gravitational_difference">,
+    filtered_angular_velocity: AdditionalOutput<Vector3<Robot>, "filtered_angular_velocity">,
+    filtered_linear_acceleration: AdditionalOutput<Vector3<Robot>, "filtered_linear_acceleration">,
+    filtered_roll_pitch: AdditionalOutput<Vector2<Robot>, "filtered_roll_pitch">,
+    fallen_down_gravitational_difference:
+        AdditionalOutput<f32, "fallen_down_gravitational_difference">,
+    fallen_sitting_gravitational_difference:
+        AdditionalOutput<f32, "fallen_standing_gravitational_difference">,
+    fallen_up_gravitational_difference: AdditionalOutput<f32, "fallen_up_gravitational_difference">,
+    upright_gravitational_difference: AdditionalOutput<f32, "upright_gravitational_difference">,
+    difference_to_sitting: AdditionalOutput<f32, "difference_to_sitting">,
+    falling_direction: AdditionalOutput<Option<Direction>, "falling_direction">,
+    fallen_direction: AdditionalOutput<Option<Kind>, "fallen_direction">,
```

Comment:
these are all in the cycler's global scope. To reduce duplication, you can also come up with a contextual struct grouping all these outputs